common/lib/dependabot/dependency_group.rb

```diff
@@ -4,17 +4,28 @@
 
 module Dependabot
   class DependencyGroup
-    attr_reader :name, :rules, :dependencies
+    attr_reader :name, :rules, :dependencies, :id
 
     def initialize(name:, rules:)
       @name = name
       @rules = rules
       @dependencies = []
+      @id = id_hash
     end
 
     def contains?(dependency)
       @dependencies.include?(dependency) if @dependencies.any?
       rules.any? { |rule| WildcardMatcher.match?(rule, dependency.name) }
     end
+
+    private
+
+    def id_hash
+      { "name" => serialized_group_name }
+    end
```

Comment:
```suggestion
    def to_h
      { "name" => serialized_group_name }
    end
    
    private
```

I think it would be more idiomatic to just make this a type-casting/serialise method and avoid bringing in the term `id` at all in the event we find we do need it future.